### ISSUE TITLE
Field with multiple inputs gets phantom entries as repeatable

### ISSUE THREAD

**Issue by davekellam** [WebDevStudios/CMB2#373]
I was attempting to do something similar to to the address field snippet sample, using a field with multiple text input boxes. It will save multiple entries just fine, but it seems that a phantom entry gets added on every save (even the initial one), so they can start to add up. The address snippet exhibits the same behaviour. Not entirely sure if it's a bug in the my code/the sample, or an issue in general with cmb2. It might be related to #117.

**Comment by jtsternberg** [comment]
Can you duplicate the issue when using the address field snippet example?

**Comment by davekellam** [comment]
Yes. When the address is set to repeatable, after the initial save, it will save an empty entry each time.

![screen shot 2015-07-15 at 12 28 19 pm](https://cloud.githubusercontent.com/assets/60804/8703788/2422e9a8-2aed-11e5-8d14-0d74026f49b1.png)

**Comment by davekellam** [comment]
Revisited the problem this afternoon, and figured out that it's a problem with the sample code (at least I think so, there could be some larger change needed).

I was able to prevent the saving by doing an empty check against one of the fields in the sanitize function: 
````php
	foreach ( $meta_value as $key => $val ) {
		if ( '' == $val['title'] ) {
			unset( $meta_value[$key] );
		} else {
			$meta_value[ $key ] = array_map( 'sanitize_text_field', $val );
		}
	}

	return $meta_value;
````

Imagine can be closed, and the discussion moved to the sample code repo. Don't know if there's a way to do a more generic check, this relies on one of the fields being `title`;

**Comment by tonezzz** [comment]
Thank you very much davekellam, your work around has just saved my life.  I've been researching on this exact issue since yesterday.

**Comment by Steviant** [comment]
@davekellam I've found that workaround to be problematic, preventing some non-empty fields from saving. Does that happen with your implementation too?

**Comment by natebeaty** [comment]
Had the same issue and ended up using this:

```
    foreach ( $meta_value as $key => $val ) {
        if ( empty( implode( '', $val ) ) ) {
            unset( $meta_value[$key] );
        } else {
            $meta_value[ $key ] = array_map( 'sanitize_text_field', $val );
        }
    }
    return $meta_value;
```

Thanks for the pointer, @davekellam.
